packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -236,8 +219,8 @@
      */
     @Watch('queryPreviewResults.status')
     emitLoad(status: RequestStatus | undefined): void {
-      if (status === 'success') {
-        this.$emit(this.results?.length ? 'load' : 'error', this.query);
```

Comment:
No need to check the length of the results here, as this scenario is handled in the `queryPreviewResults` getter

Reply:
There is something that feels not right in this approach

Reply:
The `load` event should be emitted when the results are loaded, not when they're loading

![image](https://github.com/empathyco/x/assets/7124620/3868c67b-da92-4ae7-b1f4-1412ec26b830)

In this code you're signaling `QueryPreviewList` to flag the results for a query as loaded when they're not.

Reply:
TRUE